dbms/src/Interpreters/Context.cpp

```diff
@@ -1650,9 +1650,8 @@ bool Context::initializeGlobalStoragePoolIfNeed(const PathPool & path_pool)
     auto lock = getLock();
     if (shared->global_storage_pool)
     {
-        // Can't init GlobalStoragePool twice.
-        // otherwise the pagestorage instances in `StoragePool` for each table won't be updated and cause unexpected problem.
-        throw Exception("GlobalStoragePool has already been initialized.", ErrorCodes::LOGICAL_ERROR);
+        // GlobalStoragePool may be initialized many times in some test cases for restore.
+        LOG_WARNING(shared->log, "GlobalStoragePool has already been initialized.");
```

Comment:
Do the ut rely on this?

Reply:
> Do the ut rely on this?

Yes. Each UT will create its own data path, so the global storage pool need to be re-inited.
https://github.com/pingcap/tiflash/blob/64a747a179d736475a66e02d926efbab5a133c32/dbms/src/TestUtils/TiFlashTestEnv.cpp#L96